File: cpp/CinemaFilm.cpp
#include <string>

// class to represent a cinema film
class CinemaFilm {

// private attributes
private:
	std::string filmCode;
	std::string filmTitle;
	std::string filmGenre;
	int filmDuration;
	int ticketPrice;
	int filmRating;

public:
	// empty constructor
	CinemaFilm() {
	}

	// constructor with parameters
	CinemaFilm(std::string filmCode, std::string filmTitle, std::string filmGenre,
			int filmDuration, int ticketPrice, int filmRating) {
		this->filmCode = filmCode;
		this->filmTitle = filmTitle;
		this->filmGenre = filmGenre;
		this->filmDuration = filmDuration;
		this->ticketPrice = ticketPrice;
		this->filmRating = filmRating;
	}

	// filmCode getter and setter
	std::string getFilmCode() {
		return filmCode;
	}

	void setFilmCode(std::string filmCode) {
		this->filmCode = filmCode;
	}

	// filmTitle getter and setter
	std::string getFilmTitle() {
		return filmTitle;
	}

	void setFilmTitle(std::string filmTitle) {
		this->filmTitle = filmTitle;
	}

	// filmGenre getter and setter
	std::string getFilmGenre() {
		return filmGenre;
	}

	void setFilmGenre(std::string filmGenre) {
		this->filmGenre = filmGenre;
	}

	// filmDuration getter and setter
	int getFilmDuration() {
		return filmDuration;
	}

	void setFilmDuration(int filmDuration) {
		this->filmDuration = filmDuration;
	}

	// ticketPrice getter and setter
	int getTicketPrice() {
		return ticketPrice;
	}

	void setTicketPrice(int ticketPrice) {
		this->ticketPrice = ticketPrice;
	}

	// filmRating getter and setter
	int getFilmRating() {
		return filmRating;
	}

	void setFilmRating(int filmRating) {
		this->filmRating = filmRating;
	}
};

File: python/CinemaFilm.py
class CinemaFilm:
    # constructor with parameters
    def __init__(self, filmCode, filmTitle, filmGenre, filmDuration, ticketPrice, filmRating):
        self.filmCode = filmCode
        self.filmTitle = filmTitle
        self.filmGenre = filmGenre
        self.filmDuration = filmDuration
        self.ticketPrice = ticketPrice
        self.filmRating = filmRating

    # filmCode getter and setter
    def getFilmCode(self):
        return self.filmCode

    def setFilmCode(self, filmCode):
        self.filmCode = filmCode

    # filmTitle getter and setter
    def getFilmTitle(self):
        return self.filmTitle

    def setFilmTitle(self, filmTitle):
        self.filmTitle = filmTitle

    # filmGenre getter and setter
    def getFilmGenre(self):
        return self.filmGenre

    def setFilmGenre(self, filmGenre):
        self.filmGenre = filmGenre

    # filmDuration getter and setter
    def getFilmDuration(self):
        return self.filmDuration

    def setFilmDuration(self, filmDuration):
        self.filmDuration = filmDuration

    # ticketPrice getter and setter
    def getTicketPrice(self):
        return self.ticketPrice

    def setTicketPrice(self, ticketPrice):
        self.ticketPrice = ticketPrice

    # filmRating getter and setter
    def getFilmRating(self):
        return self.filmRating

    def setFilmRating(self, filmRating):
        self.filmRating = filmRating
